client/src/components/ProgressBar/progress-bar.tsx

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { createSelector } from 'reselect';
+import { TFunction, withTranslation } from 'react-i18next';
+import {
+  challengeMetaSelector,
+  currentBlockIdsSelector,
+  completedChallengesInBlockSelector,
+  completedPercentageSelector
+} from '../../templates/Challenges/redux/selectors';
+import { certMap } from '../../resources/cert-and-project-map';
+import ProgressBarInner from './progress-bar-inner';
+
+const mapStateToProps = createSelector(
+  currentBlockIdsSelector,
+  challengeMetaSelector,
+  completedChallengesInBlockSelector,
+  completedPercentageSelector,
+  (
+    currentBlockIds: string[],
+    {
+      id,
+      block,
+      superBlock
+    }: {
+      id: string;
+      block: string;
+      superBlock: string;
+    },
+    completedChallengesInBlock: number,
+    completedPercent: number
+  ) => ({
+    currentBlockIds,
+    id,
+    block,
+    superBlock,
+    completedChallengesInBlock,
+    completedPercent
+  })
+);
+
+type StateProps = ReturnType<typeof mapStateToProps>;
+
+interface ProgressBarProps extends StateProps {
+  t: TFunction;
+}
+function ProgressBar({
+  currentBlockIds,
+  block,
+  id,
+  superBlock,
+  completedChallengesInBlock,
+  completedPercent,
+  t
+}: ProgressBarProps): JSX.Element {
+  const blockTitle = t(`intro:${superBlock}.blocks.${block}.title`);
+  const isCertificationProject = certMap.some(cert => {
+    // @ts-expect-error If `projects` does not exist, no consequences
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call
+    return cert.projects?.some((project: { id: string }) => project.id === id);
```

Comment:
This is probably unrelated to this PR, but we are here.

```suggestion
    if ('projects' in cert) {
      return cert.projects?.some(
        (project: { id: string }) => project.id === id
      );
    }
```

Reply:
You wouldn't need optional chaining there, then.

Reply:
_You always need more optional chaining_

Aside: It might be useful to have a lint rule catching unnecessary things like that.